youtube_dl/extractor/cba.py

```diff
@@ -0,0 +1,104 @@
+        }
+    }
+    _FORMATS = {
+        'audio/ogg': {'id': '1', 'ext': 'ogg', 'preference': 100},
+        'audio/mpeg': {'id': '2', 'ext': 'mp3', 'preference': 50}
+    }
+    _API_KEY = None
+
+    def __init__(self, *args, **kwargs):
+        try:
+            self._API_KEY = os.environ["CBA_API_KEY"]
```

Comment:
Hm ok - i will drop the environment variable.
Concerning the API keys: this is the peferred way by the people who run the platform. Of course there would be the possibility to mimick the browser behaviour but at least for us (a community radio station which has an API key) it is of course much less error prone to just use the key.
I will take a look into how to download the file via the flash player. Don't know whether the platfrom operators will like this though.